queries.sql

```diff
@@ -57,4 +57,164 @@ GROUP BY species;
+JOIN owners ON animals.owner_id = owners.id
+LEFT JOIN escapes ON animals.id = escapes.animal_id
+WHERE owners.name = 'Dean Winchester' AND animals.escape_attempts = 0;
+
+SELECT owners.full_name, COUNT(*) as count
+FROM owners
+JOIN animals ON owners.id = animals.owner_id
+GROUP BY owners.full_name
+ORDER BY count DESC
+LIMIT 1;
+
+SELECT * FROM animals;
+
+
+START TRANSACTION;
+SET SQL_SAFE_UPDATES = 0;
+UPDATE animals SET species = 'unspecified';
+SELECT * FROM animals;  -- verify that the change was made
+ROLLBACK;
+SELECT * FROM animals;  -- verify that the change was rolled back
+SET SQL_SAFE_UPDATES = 0;
+SELECT * FROM animals;
+
+START TRANSACTION;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species = '';
+COMMIT;
+SELECT * FROM animals;
+SELECT * FROM animals WHERE species = 'pokemon';
+
+START TRANSACTION;
+DELETE FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+START TRANSACTION;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT sp;
+UPDATE animals SET weight_kg = -1 * weight_kg;
+ROLLBACK TO sp;
+UPDATE animals SET weight_kg = -1 * weight_kg WHERE weight_kg < 0;
+COMMIT;
+SELECT * FROM animals;
+
+SELECT COUNT(*) FROM animals;
+
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+
+SELECT AVG(weight_kg) FROM animals;
+
+SELECT neutered, AVG(escape_attempts) as avg_escape_attempts
+FROM animals
+GROUP BY neutered
+ORDER BY avg_escape_attempts DESC
+LIMIT 1;
+
+SELECT species, MIN(weight_kg) as min_weight, MAX(weight_kg) as max_weight
+FROM animals
+GROUP BY species;
+
+SELECT species, AVG(escape_attempts) as avg_escape_attempts
+FROM animals
+WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'
+GROUP BY species;
+
+
+SELECT * FROM animals;
+
+CREATE TABLE owners (
+    id INTEGER NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    full_name VARCHAR(100),
+    age INTEGER
+);
+CREATE TABLE species (
+    id INTEGER NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    name VARCHAR(50)
+);
+
+ALTER TABLE animals
+    DROP COLUMN species,
+    ADD COLUMN species_id INTEGER,
+    ADD COLUMN owner_id INTEGER,
+    MODIFY COLUMN id INTEGER NOT NULL AUTO_INCREMENT;
+
+ALTER TABLE animals
+    ADD CONSTRAINT fk_species_id FOREIGN KEY (species_id)
+        REFERENCES species(id),
+    ADD CONSTRAINT fk_owner_id FOREIGN KEY (owner_id)
+        REFERENCES owners(id);
+
+-- find owner_ids
+SELECT id, full_name FROM owners;
+
+SELECT animals.name
+FROM animals
+JOIN visits ON animals.id = visits.animal_id
+JOIN vets ON visits.vet_id = vets.id
+WHERE vets.name = 'William Tatcher'
+ORDER BY visits.date_of_visit DESC
+LIMIT 1;
+
+SELECT COUNT(DISTINCT visits.animal_id)
+FROM visits
+JOIN vets ON visits.vet_id = vets.id
+WHERE vets.name = 'Stephanie Mendez';
+
+SELECT vets.name, specialties.name
+FROM vets
+LEFT JOIN specializations ON vets.id = specializations.vet_id
+LEFT JOIN species ON specializations.species_id = species.id
+ORDER BY vets.name, specialties.name;
```

Comment:
- [ Required ] I think the field `specialities.name` should be `species.name` because  your `schema.sql` file does not have a table with the name `specialities`